src/components/SignupForm.js

```diff
@@ -0,0 +1,240 @@
+import Timer from './Timer';
+import { useMediaQuery } from 'react-responsive';
+
+const SignupForm = () => {
+
+    const navigate = useNavigate();
+    const [certificationNum, setCertificationNum] = useState('');
+    const [checkPassword, setCheckPassword] = useState('');
+    const [carNumber, setCarNumber] = useState('');
+    const [timeover, setTimeover] = useState(false);
+    const [completeCertification, setCompleteCertification] = useState(false);
+    const [timerOn, setTimerOn] = useState(false);
+
+    const [signupInfo, setSignupInfo ] = useState({
+        email : '',
+        password : '',
+        name : '', 
+        auth : 'USER'
+    });
+
+    const handleSignupData = (e) => {
+        
+        if(e.target.name === 'certification') {
+            setCertificationNum(e.target.value);
+        }
+        else if (e.target.name === 'checkPassword') {
+            setCheckPassword(e.target.value);
+        }
+        else if (e.target.name === 'carNumber') {
+            setCarNumber(e.target.value);
+        }
+        else {
+            setSignupInfo({
+                ...signupInfo, 
+                [e.target.name]: e.target.value
+            });
+        }
+
+    };
+
+    // 이메일 인증 요청 통신
+    const handleCertification = () => {
+
+        if(!isValidEmail(signupInfo.email)) {
+            alert('이메일을 형식에 맞게 입력해 주세요.');
+            return;
+        }
+        
+        axios.post(`${process.env.REACT_APP_PROXY}/email`, {
+            'email': signupInfo.email
+        })
+        .then((res) => {
+            // 이메일 인증 성공
+            alert('입력한 이메일을 확인해주세요.');
+            setTimerOn(true);
+        })
+        .catch((err) => {
+            // 이메일 인증 실패
+            console.log(err);
+            alert(err.response.data.description);
+        });
+    }
+
+    // 인증번호 재발송 요청 통신
+    const handleReCertification = () => {
+
+        axios.patch(`${process.env.REACT_APP_PROXY}/email`, {
+            'email': signupInfo.email
+        })
+        .then((res) => {
+            // 재발급 성공
+            console.log(res);
+            alert('입력한 이메일을 확인해주세요.');
+            setTimerOn(true);
+        })
+        .catch((err) => {
+            // 재발급 실패
+            console.log(err);
+            setTimeover(true);
+        });
+    }
+
+    // 인증번호 입력 확인 통신
+    const handleCheckCertifNum = () => {
+        
+        axios.get(`${process.env.REACT_APP_PROXY}/email?code=${certificationNum}&email=${signupInfo.email}`)
+        .then((res) => {
+            // 인증번호 일치
+            if(res.data) {
+                alert('인증되었습니다.');
+                setCompleteCertification(true);
+                setTimerOn(false);
+            }
+            else {
+                // 인증번호 불일치
+                alert('인증번호가 만료되었습니다. 재발급해주세요.');
+                setTimeover(true);
+                setTimerOn(false);
+            }
+        })
+        .catch((err) => {
+            // // 인증번호 불일치
```

Comment:
If there is an error, it would be better to include a guide text.

For example, 'The network is unstable'.